plutus-metatheory/src/Type/RenamingSubstitution.lagda.md

```diff
@@ -25,20 +25,35 @@ open _⊢⋆_
 
 ## Type renaming
 
-A type renaming is a mapping of type variables to type variables.
+Types can contain functions and as such are subject to a nontrivial equality relation. To
+explain the computation equation (the β-rule) we need to define substitution for a single
+type variable in a type. A type renaming is a function from type variables in one context to
+type variables in another. We only need the ability to
+introduce new variable on the right hand side of the context. The simplicity of the
+definition makes it easy to work with and we get some properties for free that we would
+have to pay for with a first order representation, such as not needing to define a lookup
+function, and we inherit the properties of functions provided by η-equality, such as
+associativity of composition, for free. Note that even though
+renamings are functions we do not require our metatheory (Agda’s type system) to support
+functional extensionality. We only ever need to make use of an equation
+between renamings on a point (a variable) and therefore need only a pointwise version
+of equality on functions to work with equality of renamings and substitutions.
 
 ```
 Ren : Ctx⋆ → Ctx⋆ → Set
 Ren Φ Ψ = ∀ {J} → Φ ∋⋆ J → Ψ ∋⋆ J
 ```
 
-Let `ρ` range of renamings.
+Let `ρ` range over renamings.
 ```
 variable
   ρ ρ' : Ren Φ Ψ
 ```
 
-Extending a type renaming — used when going under a binder.
+As we are going to push renamings through types we need to be able to push them
+under a binder. To do this safely the newly bound variable should remain untouched and
+other renamings should be shifted by one to accommodate this. (Note: this is
+called `lift⋆` in the paper).
```

Comment:
In which paper?

Reply:
The answer is: the ["Fun and Profit" paper](https://ci.iog.io/build/1230848/download/1/paper.pdf) but I wasn't sure whether to reference it explicitly in the Agda, since I'd rather not have a big (and identical) bibliography in every file. I suppose it is markdown, I could do what I've just done here!